### PR TITLE
FISH-233 Multiple Path Traversal security issues

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

File: jsf-ri/src/main/java/com/sun/faces/application/resource/ResourceManager.java [hunk 1]
```diff
@@ -590,6 +590,15 @@ ResourceInfo findResource(String resourceId) {
     private String getLocalePrefix(FacesContext context) {
 
         String localePrefix = null;
+        
+        localePrefix = context.getExternalContext().getRequestParameterMap().get("loc");
+
+        if (localePrefix != null && !nameContainsForbiddenSequence(localePrefix)) {
+            return localePrefix;
+        } else {
```
Comment (rdebusscher) [inline jsf-ri/src/main/java/com/sun/faces/application/resource/ResourceManager.java:598]: else branch not needed as `localePrefix` already null.

**Review by rdebusscher — CHANGES_REQUESTED** [review]
Minor comment to update PR

File: jsf-ri/src/main/java/com/sun/faces/application/resource/ResourceManager.java (hunk 1 above)
Comment (MeroRai) [reply jsf-ri/src/main/java/com/sun/faces/application/resource/ResourceManager.java:598]: I cherry-picked it from [upstream](https://github.com/eclipse-ee4j/mojarra/pull/4629/files#diff-3c552d4607f9ebd9a197fb6c464aabdeL569). I assumed there might be a reason for setting it to null. Also, the if statement is looking at 2 conditions to not just if `localePrefix` is `null`.

Comment (rdebusscher) [reply jsf-ri/src/main/java/com/sun/faces/application/resource/ResourceManager.java:598]: Yep, my mistake, variable gets reassigned in the mean time. Thx

**Review by rdebusscher — APPROVED** [review]

**Pull request merged**